src/script/script.cpp

```diff
@@ -227,6 +227,14 @@ bool CScript::IsWitnessProgram(int& version, std::vector<unsigned char>& program
         return false;
     }
     if ((size_t)((*this)[1] + 2) == this->size()) {
```

Comment:
Might as well return this directly.

Reply:
I like that style too, but held back for the sake of simplifying review. Here's my choice if I'm to simplify the method:

```cpp
static constexpr bool IsOP_N(const unsigned char op)
{
    return (op == OP_0 || (op >= OP_1 && op <= OP_16));
}

// A witness program is any valid CScript that consists of a 1-byte push opcode
// followed by a data push between 2 and 40 bytes.
bool CScript::IsWitnessProgram() const
{
    return ((this->size() >= 4 && this->size() <= 42) &&
            IsOP_N((*this)[0]) &&
            (size_t)((*this)[1] + 2) == this->size());
}
```

Incidentally, this matches the style of `IsPayToWitnessScriptHash` and `IsPayToScriptHash`, above. Thoughts?

Reply:
@Empact looks good, I'd ACK that.